internal/pkg/crypto/integration_test.go

```diff
@@ -0,0 +1,88 @@
+package crypto
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+const port = "8181"
+
+func TestCAAndCertificateCreationInAHttpsServer(t *testing.T) {
```

Comment:
If we want to make sure the certificate being issued is valid, we could use the `Certificate.Verify` function, like done inside of the [example](https://pkg.go.dev/crypto/x509#Certificate.Verify) provided by the std library.
Using a http server is overly complicated IMHO.

Reply:
I see. Indeed, that makes sense. As I said in a previous comment, as this package is a children from other packages in the `internal/pkg` directory. Which includes the test cases as well. Let me adapt the tests

Reply:
Reminder to update the test case